differ/differ.go

```diff
@@ -1144,30 +1170,56 @@ func closeDiffer(storage *DBStorage) {
 	log.Info().Msg(separator)
 }
 
+// setupFiltersAndThresholds function setup both techniques that can be used to
+// filter messages sent to targets (Notification backend and ServiceLog at this moment):
+//     1. filter based on likelihood, impact, severity, and total risk
+//     2. filter based on rule type that's identified by tags
 func setupFiltersAndThresholds(config *conf.ConfigStruct) {
-	kafkaEventThresholds.Likelihood = conf.GetKafkaBrokerConfiguration(config).LikelihoodThreshold
-	kafkaEventThresholds.Impact = conf.GetKafkaBrokerConfiguration(config).ImpactThreshold
-	kafkaEventThresholds.Severity = conf.GetKafkaBrokerConfiguration(config).SeverityThreshold
-	kafkaEventThresholds.TotalRisk = conf.GetKafkaBrokerConfiguration(config).TotalRiskThreshold
-	kafkaEventFilter = conf.GetKafkaBrokerConfiguration(config).EventFilter
+	kafkaBrokerConfiguration := conf.GetKafkaBrokerConfiguration(config)
+
+	kafkaEventThresholds.Likelihood = kafkaBrokerConfiguration.LikelihoodThreshold
+	kafkaEventThresholds.Impact = kafkaBrokerConfiguration.ImpactThreshold
+	kafkaEventThresholds.Severity = kafkaBrokerConfiguration.SeverityThreshold
+	kafkaEventThresholds.TotalRisk = kafkaBrokerConfiguration.TotalRiskThreshold
+	kafkaEventFilter = kafkaBrokerConfiguration.EventFilter
 
 	if kafkaEventFilter == "" {
-		err := fmt.Errorf("Configuration problem")
+		err := fmt.Errorf(configurationProblem)
 		log.Err(err).Msg(eventFilterNotSetMessage)
 		os.Exit(ExitStatusEventFilterError)
 	}
 
-	serviceLogEventThresholds.Likelihood = conf.GetServiceLogConfiguration(config).LikelihoodThreshold
-	serviceLogEventThresholds.Impact = conf.GetServiceLogConfiguration(config).ImpactThreshold
-	serviceLogEventThresholds.Severity = conf.GetServiceLogConfiguration(config).SeverityThreshold
-	serviceLogEventThresholds.TotalRisk = conf.GetServiceLogConfiguration(config).TotalRiskThreshold
-	serviceLogEventFilter = conf.GetServiceLogConfiguration(config).EventFilter
+	kafkaFilterByTagsEnabled = kafkaBrokerConfiguration.TagFilterEnabled
+	kafkaTagsSet = kafkaBrokerConfiguration.TagsSet
+
+	if kafkaTagsSet == nil {
```

Comment:
if `kafkaFilterByTagsEnabled` is false we can have `kafkaTagsSet == nil`

Reply:
good comment, lemme fix this one!

Reply:
fixed!